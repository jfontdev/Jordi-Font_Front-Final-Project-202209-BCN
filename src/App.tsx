import { Route, Routes } from "react-router-dom";
import Layout from "./components/Layout/Layout";
import CreateReview from "./pages/CreateReview/CreateReview";
import FilmDetail from "./pages/FilmDetail/FilmDetail";
import HomePage from "./pages/Homepage/HomePage";
import Register from "./pages/Register/Register";

const App = () => {
  return (
    <Routes>
      <Route element={<Layout />}>
        <Route path="/" element={<HomePage />}></Route>
        <Route path="/register" element={<Register />}></Route>
        <Route path="/film-detail" element={<FilmDetail />}></Route>
        <Route path="/create-review" element={<CreateReview />}></Route>
      </Route>
    </Routes>
  );
};

export default App;
