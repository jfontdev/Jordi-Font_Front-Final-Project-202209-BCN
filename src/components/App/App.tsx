import { Route, Routes } from "react-router-dom";
import Layout from "../Layout/Layout";
import CreateReview from "../../pages/CreateReview/CreateReview";
import HomePage from "../../pages/Homepage/HomePage";
import Register from "../../pages/Register/Register";
import React, { Suspense } from "react";
import Loading from "../Loading/Loading";

const FilmDetail = React.lazy(
  () => import("../../pages/FilmDetail/FilmDetail")
);

const Detail = React.lazy(() => import("../../pages/Detail/Detail"));

const App = () => {
  return (
    <Suspense fallback={<Loading />}>
      <Routes>
        <Route element={<Layout />}>
          <Route path="/" element={<HomePage />}></Route>
          <Route path="/register" element={<Register />}></Route>
          <Route path="/film-detail" element={<FilmDetail />}></Route>
          <Route path="/create-review" element={<CreateReview />}></Route>
          <Route path="/review-detail/:id" element={<Detail />}></Route>
        </Route>
      </Routes>
    </Suspense>
  );
};

export default App;
