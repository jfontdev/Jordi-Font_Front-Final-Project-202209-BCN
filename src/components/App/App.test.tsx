import TestRenderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import App from "./App";

describe("Given an App component", () => {
  describe("When its rendered", () => {
    test("With path '/*', it should render NotFoundPage'", () => {
      const expectedApp = TestRenderer.create(
        <MemoryRouter initialEntries={["/*"]}>
          <App />
        </MemoryRouter>
      );
      expect(expectedApp).toMatchSnapshot();
    });
  });
});
