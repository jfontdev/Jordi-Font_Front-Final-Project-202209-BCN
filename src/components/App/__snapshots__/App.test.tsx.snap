// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When its rendered With path '/*', it should render NotFoundPage' 1`] = `null`;
